src/qt/masternodelist.cpp

```diff
@@ -646,3 +667,63 @@ void MasternodeList::ShowQRCode(std::string strAlias)
     dialog->setInfo(strWindowtitle, QString::fromStdString(strMNPrivKey), strHTML, strQRCodeTitle);
     dialog->show();
 }
+
+CDeterministicMNCPtr MasternodeList::GetSelectedDIP3MN()
+{
+    std::string strAddress;
+    {
+        LOCK(cs_dip3list);
+
+        QItemSelectionModel* selectionModel = ui->tableWidgetMasternodesDIP3->selectionModel();
+        QModelIndexList selected = selectionModel->selectedRows();
+
+        if (selected.count() == 0) return nullptr;
+
+        QModelIndex index = selected.at(0);
+        int nSelectedRow = index.row();
+        strAddress = ui->tableWidgetMasternodesDIP3->item(nSelectedRow, 0)->text().toStdString();
+    }
+    CService addr;
+    if (!Lookup(strAddress.c_str(), addr, Params().GetDefaultPort(), false)) {
+        return nullptr;
+    }
+    auto mnList = deterministicMNManager->GetListAtChainTip();
+    return mnList.GetUniquePropertyMN(addr);
+}
+
+void MasternodeList::on_extraInfoDIP3_clicked()
+{
+    auto dmn = GetSelectedDIP3MN();
+    if (!dmn) {
+        return;
+    }
+
+    UniValue json(UniValue::VOBJ);
+    dmn->ToJson(json);
+
+    // Title of popup window
+    QString strWindowtitle = tr("Additional information for DIP3 Masternode %1").arg(QString::fromStdString(dmn->proTxHash.ToString()));
+    QString strText = QString::fromStdString(json.write(2));
+
+    QMessageBox::information(this, strWindowtitle, strText);
+}
+
+void MasternodeList::on_copyProTxHash_clicked()
+{
+    auto dmn = GetSelectedDIP3MN();
+    if (!dmn) {
+        return;
+    }
+
+    QApplication::clipboard()->setText(QString::fromStdString(dmn->proTxHash.ToString()));
+}
+
+void MasternodeList::on_copyCollateralTxHash_clicked()
+{
+    auto dmn = GetSelectedDIP3MN();
+    if (!dmn) {
+        return;
+    }
+
+    QApplication::clipboard()->setText(QString::fromStdString(dmn->collateralOutpoint.ToStringShort()));
```

Comment:
Technically speaking, this is not a hash but an outpoint i.e. menu text/function name is misleading.